### ISSUE TITLE
[Bug] i.image.bathymetry needs corrections to the R scripting calls to run all options

### ISSUE THREAD

**Issue by newcomb-d** [OSGeo/grass-addons#289]
**Describe the bug**
i.image.bathymetry can call the  R GWModel program with three options, Adaptive 
Gaussian , Bisquare, and Fixed Gaussian.  If there are issues running bisquare or adaptive, the script fails over  to Fixed.  With versions 3.5.3 and 3.6.3 of R, the current call to the R package raster fails and the script runs GWR in Fixed Gaussian mode , no matter what option is given in the command

**To Reproduce**

1. Load multispectral imagery into GRASS ( Sentinel 2 bands 2,3,4,8, and 11) for an area of interest with a water body for which bathymetry is needed.  Also load know depth points, tide level, and a polygon area of interest.
2. Set region to match Area of interest with resolution matching satellite imagery.
3. Run script from the command line. ( Example) i.image.bathymetry  blue_band=T18STF_20191212T155641_B02_10m green_band=T18STF_20191212T155641_B03_10m red_band=T18STF_20191212T155641_B04_10m nir_band=T18STF_20191212T155641_B08_10m band_for_correction=T18STF_20191212T155641_B11_20m calibration_points=bathy_points calibration_column=depth depth_estimate=depth_out area_of_interest=lake_Shoreline_trim_buffer

Attaching package: 'raster'

The following object is masked from 'package:data.table':

    shift

Warning message:
package 'raster' was built under R version 3.5.3
Error in .read_rast_non_plugin(vname = vname, NODATA = NODATA, driverFileExt = driverFileExt,  :
  no stars import yet
Calls: readRAST -> tryCatch -> tryCatchList -> .read_rast_non_plugin
Execution halted
Integer outflow...
Running Fixed-GWR using gaussian kernel...

**Expected behavior**
The default should use the Adaptive Guassian kernel

Calculating optimal bandwidth using gaussian kernel..
Adaptive bandwidth: 217 CV score: 1608.003
Adaptive bandwidth: 142 CV score: 1511.212
Adaptive bandwidth: 94 CV score: 1430.953
Adaptive bandwidth: 66 CV score: 1335.32
Adaptive bandwidth: 47 CV score: 1271.241
Adaptive bandwidth: 37 CV score: 1249.903
Adaptive bandwidth: 29 CV score: 1231.113
Adaptive bandwidth: 25 CV score: 1217.598
Adaptive bandwidth: 22 CV score: 1212.767
Adaptive bandwidth: 20 CV score: 1207.13
Adaptive bandwidth: 19 CV score: 1208.674
Adaptive bandwidth: 21 CV score: 1211.075
Adaptive bandwidth: 20 CV score: 1207.13

Calculating euclidean distance

Running A-GWR using gaussian kernel
Reading input data...
 100%
146789 points found in input file
Writing to output raster map...
 100%

**Screenshots**

**System description (please complete the following information):**

- Operating System: Windows 10
- GRASS GIS version 7.8.3
- R versions 3.5.3 and 3.6.3

**Additional context**

**Comment by neteler** [comment]
Please consider to edit the Python script directly here and submit the changes as a pull request:
https://github.com/OSGeo/grass-addons/blob/master/grass7/imagery/i.image.bathymetry/i.image.bathymetry.py

**Comment by newcomb-d** [comment]
Sorry for the delay.  Working with external user who can submit via web interface , but is still getting up to speed on command line git , which seems to be the project preference.

Doug


________________________________

**Comment by tmszi** [comment]
> The script was then able to get past this error. However, in line 293
> predict = g.read_command("g.tempfile", pid=os.getpid()).strip() + '.txt'
> a temporary file is written to hold the results of the GWR analysis.
> On windows, this leads to a filename like this:
> D:\grass\lake\u18wgs84\bathy.tmp/unknown\13752.5.txt , which throws an escape character error in R on windows.

This is bug in the G__temp_element() function (line [121)](https://grass.osgeo.org/programming7/tempfile_8c_source.html#l00113).

**Comment by neteler** [comment]
> > On windows, this leads to a filename like this:
> > D:\grass\lake\u18wgs84\bathy.tmp/unknown\13752.5.txt , which throws an escape character error in R on windows.
> 
> This is bug in the G__temp_element() function (line [121](https://grass.osgeo.org/programming7/tempfile_8c_source.html#l00113)).

Probably `G_convert_dirseps_to_host()` is missing? See
https://grass.osgeo.org/programming7/paths_8c.html#a0bd23f09c109997ed8ddc25c5d650698

**Comment by tmszi** [comment]
> > > On windows, this leads to a filename like this:
> > > D:\grass\lake\u18wgs84\bathy.tmp/unknown\13752.5.txt , which throws an escape character error in R on windows.
> > 
> > 
> > This is bug in the G__temp_element() function (line [121](https://grass.osgeo.org/programming7/tempfile_8c_source.html#l00113)).
> 
> Probably `G_convert_dirseps_to_host()` is missing? See
> https://grass.osgeo.org/programming7/paths_8c.html#a0bd23f09c109997ed8ddc25c5d650698

Yes you are right or use [HOST_DIRSEP](https://grass.osgeo.org/programming7/gis_8h_source.html#l00210). 

Yesterday I prepared patch (need test on MS Win), but I have trouble with Cross-Compilation GRASS GIS for Windows OS on GNU/Linux. Otherwise I would have opened a new PR.

**Comment by neteler** [comment]
Right, `HOST_DIRSEP` is widely used in GRASS-core.

**Comment by neteler** [comment]
> Yes you are right or use [HOST_DIRSEP](https://grass.osgeo.org/programming7/gis_8h_source.html#l00210).
> 
> Yesterday I prepared patch (need test on MS Win), but I have trouble with Cross-Compilation (environment) GRASS GIS for Windows OS on GNU/Linux. Otherwise I would have opened a new PR.

While checking the entire core source code I found, besides `lib/gis/tempfile.c` two more files in which `HOST_DIRSEP` seems to be missing (unrelated but like worth to also fix in the same PR):

- `lib/display/r_raster.c` (2 times)
- `display/d.mon/list.c` (4 times)

**Comment by wenzeslaus** [comment]
Isn't the issue more that the backslash is used rather than missing? If the error is about "an escape character" in R on Windows, that suggests that the backslashes in path are not escaped or that the string in generated R code (is there some in this case?) needs to be raw. It seems to me then that if you want to change it in grass core, you need to remove the backslashes from path rather than adding more of them and use forward slashes instead.

The question also is if g.tempfile is the right tool. Is the file huge or copied to mapset non-temporary directories? If not, standard Python temporary directory object used with a `with` statement and a hardcoded file name might be a better option.

**Comment by newcomb-d** [comment]
I just noticed a similar error for r.in.wms for 7.8.7 on Windows.
